record.h

```diff
@@ -60,6 +61,10 @@ typedef struct janus_recorder {
 	volatile int header;
 	/*! \brief Whether this recorder instance can be used for writing or not */
 	volatile int writable;
+	/*! \brief Whether writing RTP packets is paused */
+	volatile int paused;
+	/*! \brief RTP switching context for rewriting RTP headers */
+	janus_rtp_switching_context context;
```

Comment:
I'm not fond of the idea of having a context as part of the recorder itself. Our plugins all have a dedicated context which they use for recordings too, even though I guess in this case we need something that's specific to the recording itself. I'll need to think about this.